src/assignment/assignment.const.ts

```diff
@@ -104,6 +123,14 @@ export const GET_ASSIGNMENT_OVERVIEW_COLUMNS = (
 	...getLastEditColumn(canEditAssignments),
 ];
 
+export const GET_ASSIGNMENT_OVERVIEW_COLUMNS = (
+	canEditAssignments: boolean | null
+): AssignmentColumn[] => [
+	...GET_ASSIGNMENT_OVERVIEW_COLUMNS_FOR_MODAL(canEditAssignments),
+	...getResponseColumn(canEditAssignments),
+	...getActionsColumn(canEditAssignments),
+];
```

Comment:
I would either use a single function and check the the last 2 columns inside the getResponseColumn and getActionsColumn functions:
```typescript 
export const GET_ASSIGNMENT_OVERVIEW_COLUMNS = (
	canEditAssignments: boolean | null
): AssignmentColumn[] => [
	{
		id: 'title',
		label: i18n.t('assignment/views/assignment-overview___titel'),
		sortable: true,
		dataType: 'string',
	},
	...getClassColumn(canEditAssignments),
	...getLabelsColumn(),
	...getTeacherColumn(canEditAssignments),
	{
		id: 'deadline_at' as AssignmentOverviewTableColumns,
		label: i18n.t('assignment/views/assignment-overview___deadline'),
		sortable: true,
		dataType: 'dateTime',
	},
	...getLastEditColumn(canEditAssignments),
	...getResponseColumn(canEditAssignments), // Not needed for "add to assignment" modal
	...getActionsColumn(canEditAssignments), // Not needed for "add to assignment" modal
];
```

Or remove the extra check inside the getResponseColumn and getActionsColumn and use both GET_ASSIGNMENT_OVERVIEW_COLUMNS and GET_ASSIGNMENT_OVERVIEW_COLUMNS_FOR_MODAL
```typescript
const getResponseColumn = (canEditAssignments: boolean | null): AssignmentColumn[] => {
	return {
		    id: 'responses' as AssignmentOverviewTableColumns,
		    label: i18n.t('assignment/assignment___respons'),
		    sortable: true,
		    dataType: 'number',
	  };
};

const getActionsColumn = (canEditAssignments: boolean | null): AssignmentColumn[] => {
	return { id: 'actions' as AssignmentOverviewTableColumns, label: '' };
};
```

But now it's a bit of both approaches